### PR TITLE
Adding MPAS-Ocean documentation

### PR DESCRIPTION
**Pull request by alicebarthel**
This includes the comprehensive MPAS-Ocean Users guide pdf (generated by @xylar), and a quick start guide.

### CODE REVIEW

**Comment by alicebarthel** [comment]
I plan to include templates of G- and C-case runscripts - which I have yet to clean up and test. Can you review the quick start guide for now?

**Comment by rljacob** [comment]
Note the linter fails.  Those need to be fixed:
```
Error: components/mpas-ocean/docs/user-guide/index.md:125:388 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1] https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md009.md
Error: components/mpas-ocean/docs/user-guide/index.md:125:230 MD033/no-inline-html Inline HTML [Element: package] https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md033.md
```

**Comment by alicebarthel** [comment]
Just reporting that I had issues with the G(MPAS-JRA1p5) case on Anvil, and both the G- and C-cases on pm. All using v3.0.0 and LR (IcoswISC30E3r5). I have not done a comprehensive survey to test all the meshes mentioned on all machines. 
@jonbob let me know if you need me to do that.

File: components/mpas-ocean/docs/user-guide/index.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+Note: the mesh should be consistent with the compset (e.g. JRA vs CORE). Additional mesh information can be found [here](https://github.com/E3SM-Project/E3SM/blob/master/cime_config/config_grids.xml).
+
+## Customizing runs
+
+### Namelist changes
+
+Without additional input, E3SM will generate the namelist file `mpaso_in` in the run directory using the default values for the compset and mesh requested.
+Namelist parameters can be changed from default values by modifying the `user_nl_mpaso` file, found in the `case_scripts` directory. This is done by entering ``[namelist option] = [changed value]`` as separate lines in the ``user_nl_mpaso`` file. All other options will remain defaults. These changes can be added at run time and will take effect in the next submission.
+
+Refer to the [MPAS-Ocean Users Guide](../tech-guide/mpas_ocean_users_guide.pdf) (Chapter 10) for a comprehensive description of the namelist parameters and the options that they correspond to.
```
Comment (mark-petersen) [inline components/mpas-ocean/docs/user-guide/index.md:69]:
```suggestion
Refer to the [MPAS-Ocean Users Guide](../tech-guide/mpas_ocean_users_guide.pdf) (Chapter 10) for a comprehensive description of the namelist parameters and the options that they correspond to. Namelist options may also be found in the code repository in the file `components/mpas-ocean/src/Registry.xml` for general flags and `components/mpas-ocean/src/tracer_groups/Registry_*.xml` for specific tracer group flags.
```

File: components/mpas-ocean/docs/user-guide/index.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+```text
+ config_GM_closure = 'constant'
+ config_gm_constant_kappa = 900
+ config_time_integrator = 'split_explicit'
+ config_am_timeseriesstatsmonthly_compute_interval = '00-00-01_00:00:00'
+```
+
+Reminder: `user_nl_mpaso` can be empty. All options not specified are defaults (given the compset and mesh). Some options (like interior restoring) require extra fields to be present in the input file.  
+
+### Configuring input and output files for MPAS-Ocean
```
Comment (mark-petersen) [inline components/mpas-ocean/docs/user-guide/index.md:82]:
```suggestion
### Configuring input and output for MPAS-Ocean using the streams.ocean file
```

File: components/mpas-ocean/docs/user-guide/index.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+<immutable_stream name="input"
+                  type="input"
+                  io_type="pnetcdf"
+                  input_interval="initial_only"
+                  filename_template="/lcrc/group/e3sm/data/inputdata/ocn/mpas-o/IcoswISC30E3r5/mpaso.IcoswISC30E3r5.20231120.nc"
+/>
+```
+
+The `mesh` filename points to the mesh file used. The `input` filename points to the file containing the ocean initial conditions (if the run type is `initial`).
+The streams file can be large, it is often useful to rely on the search function to navigate it.
```
Comment (mark-petersen) [inline components/mpas-ocean/docs/user-guide/index.md:108]:
```suggestion
The streams file can be large, it is often useful to rely on the search function to navigate it. For larger meshes (millions of horizontal cells) the flag `io_type="pnetcdf"` must be changed to `io_type="pnetcdf,cdf5"`.
```

**Review by mark-petersen — APPROVED** [review]
@alicebarthel thank you, thank you, thank you! This is exactly what we need. Here are my comments.

File: components/mpas-ocean/docs/user-guide/index.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
+
+# MPAS-Ocean Quick Start
+
+This MPAS-Ocean Quick Start Guide describes how to set up and run MPAS-Ocean within E3SM. The more detailed MPAS-Ocean Users Guide can be found [here](../tech-guide/mpas_ocean_users_guide.pdf).
```
[the remark this answers is not in the document]
Comment (xylar) [reply components/mpas-ocean/docs/user-guide/index.md:4]: @mark-petersen and @alicebarthel, please replace the PDF with the version in this comment:
https://github.com/MPAS-Dev/MPAS-Documents/pull/77#issuecomment-2081185289
That has the requested fixes.

**Comment by xylar** [comment]
@mark-petersen, can you please merge https://github.com/MPAS-Dev/MPAS-Documents/pull/77 as soon as this PR is merged?  I think it makes sense to keep it open until then in case we make further changes.

**Comment by mark-petersen** [comment]
@alicebarthel We need to add the following files:
```
components/mpas-ocean/docs/index.md
components/mpas-ocean/docs/tech-guide/index.md
components/mpas-ocean/docs/dev-guide/index.md
```
even if they are stubs with 'coming soon' on them.  That way all the components have matching file names for the links. We have external documents and papers that we can link to within the tech and dev guides.

I can add that today or tomorrow. See examples on:
https://github.com/E3SM-Project/E3SM/pull/6314/files
https://github.com/E3SM-Project/E3SM/pull/6356/files

File: components/mpas-ocean/docs/user-guide/index.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+2. [Scientifically supported compsets and meshes](#scientifically-supported-compsets-and-meshes)
+    1. [Compsets](#compsets)
+    2. [Meshes](#meshes)
+3. [Customizing runs](#customizing-runs)
+    1. [Namelist changes](#namelist-changes)
+    2. [Configuring input and output files for MPAS-Ocean](#configuring-input-and-output-files-for-mpas-ocean)
+
+## Steps to build and run MPAS-Ocean
+
+A step-by-step instruction on how to run E3SM can be found [here](https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/2309226536).
```
[the remark this answers is not in the document]
Comment (rljacob) [reply components/mpas-ocean/docs/user-guide/index.md:18]: Use this URL for the step-by-step guide: https://docs.e3sm.org/running-e3sm-guide/

**Comment by mark-petersen** [comment]
@alicebarthel I added the files in the last post and pushed them to my fork https://github.com/mark-petersen/E3SM branch `ocn/documentation` commit 8884471d12.

You can see my new pages here:
- [home page](https://github.com/mark-petersen/E3SM/blob/ocn/documentation/components/mpas-ocean/docs/index.md)
- [dev guide](https://github.com/mark-petersen/E3SM/blob/ocn/documentation/components/mpas-ocean/docs/dev-guide/index.md)
- [tech guide](https://github.com/mark-petersen/E3SM/blob/ocn/documentation/components/mpas-ocean/docs/tech-guide/index.md)

If you agree you can grab my commit and add it to this PR.

File: components/mpas-ocean/docs/user-guide/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mark-petersen) [reply components/mpas-ocean/docs/user-guide/index.md:18]: Done, thanks.

File: components/mpas-ocean/docs/user-guide/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (alicebarthel) [reply components/mpas-ocean/docs/user-guide/index.md:4]: Done! Thanks

**Comment by mark-petersen** [comment]
We would like to have this merge tomorrow morning so we are ready for the tutorial next week. If any reviewers have comments, please make them today. I gave everything a thorough read and revision, so it's OK if you don't have time.

File: components/mpas-ocean/docs/tech-guide/index.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# MPAS-Ocean Technical Guide
+
+This Technical Guide describes the governing equations, physics, and numerical discretizations of MPAS-Ocean.
+
+## Guides, Design Documents, and documentation
+
+MPAS-Ocean may be run as either a stand-alone model, or within the E3SM coupled climate model.
+The MPAS-Ocean code for both stand-alone and coupled is housed in the repository [https://github.com/E3SM-Project/E3SM](https://github.com/E3SM-Project/E3SM) within the directory components/mpas-ocean. The stand-alone executable may be built within that directory using the make command with the required libraries, as described in Chapter 1 of the [User's Guide](mpas_ocean_users_guide.pdf).
```
Comment (xylar) [inline components/mpas-ocean/docs/tech-guide/index.md:8]:
```suggestion
The MPAS-Ocean code for both stand-alone and coupled is housed in the repository [https://github.com/E3SM-Project/E3SM](https://github.com/E3SM-Project/E3SM) within the directory `components/mpas-ocean`. The stand-alone executable may be built within that directory using the make command with the required libraries, as described in Chapter 1 of the [User's Guide](mpas_ocean_users_guide.pdf).
```

**Review by xylar — APPROVED** [review]
@alicebarthel, I think this is an excellent start.  I didn't go through the example run scripts, I just don't think I will have time before this needs to go in.  But I looked through the Markdown files and saw only one minor aesthetic correction.

File: components/mpas-ocean/docs/user-guide/index.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+- Simulation compsets, resolution and name (see below)
+- Wallclock duration in `WALLTIME` and simulation duration with `STOP_OPTION` and `STOP_N`.
+
+Additional runscript examples can be found [here](https://github.com/E3SM-Project/SimulationScripts/tree/master/archive/CoupledGroup/v3.LR) for v3.LR.
+
+## Scientifically supported compsets and meshes
+
+### Compsets
+
+The compsets below are typical ocean-focused compsets supported by E3SM:
```
Comment (vanroekel) [inline components/mpas-ocean/docs/user-guide/index.md:35]: add sea ice as well here?

File: components/mpas-ocean/docs/user-guide/index.md [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+
+`GMPAS-JRA1p5` - Active ocean-sea ice configuration forced by data atmosphere based on JRA55 v1.5 (covers 63 years, 1958-2020)
+
+`GMPAS-IAF` - Active ocean-sea ice configuration forced by data atmosphere based on CORE-II (covers 62 years, 1948-2009)
+
+`GMPAS-JRA1p5-DIB-DISMF` - Active ocean-sea ice configuration forced by JRA55 v1.5 atmosphere (as above), with data iceberg and data ice-shelf melt
+
+`GMPAS-JRA1p5-DIB-PISMF` - Active ocean-sea ice configuration forced by JRA55 v1.5 atmosphere (as above), with data iceberg and data ice-shelf melt
```
Comment (vanroekel) [inline components/mpas-ocean/docs/user-guide/index.md:43]: isn't this 'prognostic ice melt'?

File: components/mpas-ocean/docs/user-guide/index.md [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+
+`CMPASO-JRA1p4` - Active ocean configuration forced by data atmosphere based on JRA55 v1.4 (covers 61 years, 1958-2018)
+
+`CMPASO-IAF` - Active ocean configuration forced by data atmosphere based on CORE-II (covers 62 years, 1948-2009)
+
+Additional compsets can be found in the [mpas-ocean `config_compsets.xml`](https://github.com/E3SM-Project/E3SM/blob/master/components/mpas-ocean/cime_config/config_compsets.xml). Note that the fully coupled compsets and their aliases can be found in the [cime allactive `config_compsets.xml`](https://github.com/E3SM-Project/E3SM/blob/master/cime_config/allactive/config_compsets.xml).
+For more information on the schemes used within MPAS-Ocean, refer to the [MPAS-Ocean Users Guide](../tech-guide/mpas_ocean_users_guide.pdf).
```
Comment (vanroekel) [inline components/mpas-ocean/docs/user-guide/index.md:50]: this seems like a jump, by schemes do you mean algorithms/parameterizations?

File: components/mpas-ocean/docs/user-guide/index.md [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+A full list of Compsets in the current repository can be listed using
+
+```text
+cd cime/scripts
+./query_config --compsets
+```
+
+### Meshes
+
+Supported meshes for G- and C-cases include:
```
Comment (vanroekel) [inline components/mpas-ocean/docs/user-guide/index.md:61]: maybe specify 'Some supported meshes'

File: components/mpas-ocean/docs/user-guide/index.md [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+
+`TL319_IcoswISC30E3r5` - Icosahedral 30 km mesh with ice shelves cavities (wISC), E3SMv3 (E3) revision r5, TL319 is the grid for JRA.
+
+`T62_IcoswISC30E3r5` - Icosahedral 30 km mesh with ice shelves cavities (wISC), E3SMv3 (E3) revision r5, T62 is the grid for CORE-II.
+
+`T62_oQU240` - Quasi uniform 2-degree ocean mesh, to be used with CORE-II only. Good for rapid testing, used in nightly testing **not** production runs.
```
Comment (vanroekel) [inline components/mpas-ocean/docs/user-guide/index.md:67]: may want to specify it is not scientifically validated

File: components/mpas-ocean/docs/user-guide/index.md [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+```
+
+## Customizing runs
+
+### Namelist changes
+
+Without additional input, E3SM will generate the namelist file `mpaso_in` in the run directory using the default values for the compset and mesh requested.
+Namelist parameters can be changed from default values by modifying the `user_nl_mpaso` file, found in the `case_scripts` directory. This is done by entering ``[namelist option] = [changed value]`` as separate lines in the ``user_nl_mpaso`` file. All other options will remain defaults. These changes can be added at run time and will take effect in the next submission.
+
+Refer to the [MPAS-Ocean Users Guide](../tech-guide/mpas_ocean_users_guide.pdf) (Chapter 10) for a comprehensive description of the namelist parameters and the options that they correspond to. Namelist options may also be found in the code repository in the file `components/mpas-ocean/src/Registry.xml` for general flags and `components/mpas-ocean/src/tracer_groups/Registry_*.xml` for specific tracer group flags.
```
Comment (vanroekel) [inline components/mpas-ocean/docs/user-guide/index.md:85]: may not be important but analysis member flags are in yet a different spot `components/mpas-ocean/src/analysis_members/Registry_*.xml`

File: components/mpas-ocean/docs/user-guide/index.md [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+
+Under each block header is the list of variables (individual variables, variable structure, or variable arrays) that will be output within the relevant stream.
+
+##### Example workflow for modifying the output fields
+
+- copy the `streams.ocean` file from the run directory to the `SourceMods` directory (see above)
+- identify the variable name for the variable of interest. You can find the variable name by searching the ocean Registry.xml (in the `src` directory) or Registry_package.xml in the `tracer` and `analysis_member` sub-directories. Note whether the variable of interest is included within a `var_array` or a `var_struct`.
+- identify the output stream of interest (e.g. monthly averages, high frequency, others). You can search for a stream name, known output filename, or output interval.
+- check whether the variable of interest is included in the `ocean.streams` file. Search for the `var name`, or the `var_array` and `var_struct` if applicable. If it is, copy the variable line from other streams into the stream of interest. If it is not included, copy it from the Registry.xml.
```
Comment (vanroekel) [inline components/mpas-ocean/docs/user-guide/index.md:145]: should we have `streams.ocean` here instead of `ocean.streams`

**Review by vanroekel — COMMENTED** [review]
Looks like a great start @alicebarthel - just a few comments on clarity.

**Comment by mark-petersen** [comment]
User's guide is now here: https://zenodo.org/records/11098080

File: components/mpas-ocean/docs/tech-guide/index.md (hunk 6 above)
Comment (mark-petersen) [inline components/mpas-ocean/docs/tech-guide/index.md:8]:
```suggestion
The MPAS-Ocean code for both stand-alone and coupled is housed in the repository [https://github.com/E3SM-Project/E3SM](https://github.com/E3SM-Project/E3SM) within the directory components/mpas-ocean. The stand-alone executable may be built within that directory using the make command with the required libraries, as described in Chapter 1 of the [MPAS-Ocean User's Guide](https://zenodo.org/records/11098080).
```

File: components/mpas-ocean/docs/tech-guide/index.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+
+The [User's Guide](mpas_ocean_users_guide.pdf) provides a description of the MPAS Framework in Part I, the Governing equations for MPAS-Ocean in Chapter 8, and describes the physics behind each term and parameterization in chapter 11.
```
Comment (mark-petersen) [inline components/mpas-ocean/docs/tech-guide/index.md:10]:
```suggestion
The [MPAS-Ocean User's Guide](https://zenodo.org/records/11098080) provides a description of the MPAS Framework in Part I, the Governing equations for MPAS-Ocean in Chapter 8, and describes the physics behind each term and parameterization in chapter 11.
```

File: components/mpas-ocean/docs/user-guide/index.md [hunk 15]
```diff
@@ -0,0 +1,210 @@
+
+# MPAS-Ocean Quick Start
+
+This MPAS-Ocean Quick Start Guide describes how to set up and run MPAS-Ocean within E3SM. The more detailed MPAS-Ocean Users Guide can be found [here](../tech-guide/mpas_ocean_users_guide.pdf).
```
Comment (mark-petersen) [inline components/mpas-ocean/docs/user-guide/index.md:4]:
```suggestion
This MPAS-Ocean Quick Start Guide describes how to set up and run MPAS-Ocean within E3SM. More details can be found in the [MPAS-Ocean User's Guide](https://zenodo.org/records/11098080),  as well as instructions on running the stand-alone ocean model.
```

File: components/mpas-ocean/docs/user-guide/index.md (hunk 9 above)
Comment (mark-petersen) [inline components/mpas-ocean/docs/user-guide/index.md:50]:
```suggestion
For more information on the schemes used within MPAS-Ocean, refer to the [MPAS-Ocean User's Guide](https://zenodo.org/records/11098080).
```

File: components/mpas-ocean/docs/user-guide/index.md (hunk 12 above)
Comment (mark-petersen) [inline components/mpas-ocean/docs/user-guide/index.md:85]:
```suggestion
Refer to the [MPAS-Ocean User's Guide](https://zenodo.org/records/11098080) (Chapter 10) for a comprehensive description of the namelist parameters and the options that they correspond to. Namelist options may also be found in the code repository in the file `components/mpas-ocean/src/Registry.xml` for general flags and `components/mpas-ocean/src/tracer_groups/Registry_*.xml` for specific tracer group flags.
```

File: components/mpas-ocean/docs/user-guide/index.md [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+    <var name="tThreshMLD"/>
+    <var name="columnIntegratedSpeed"/>
+    <var name="barotropicSpeed"/>
+    <var name="landIceFreshwaterFlux"/>
+    <var name="pressureAdjustedSSH"/>
+    <var name="atmosphericPressure"/>
+</stream>
+```
+
+A more comprehensive description of the streams options can be found in Chapter 6 of the [MPAS-Ocean Users Guide](../tech-guide/mpas_ocean_users_guide.pdf).
```
Comment (mark-petersen) [inline components/mpas-ocean/docs/user-guide/index.md:210]:
```suggestion
A more comprehensive description of the streams options can be found in Chapter 6 of the [MPAS-Ocean User's Guide](https://zenodo.org/records/11098080).
```

**Review by mark-petersen — COMMENTED** [review]
These just change to the zenodo link.

File: components/mpas-ocean/docs/user-guide/index.md (hunk 13 above)
Comment (mark-petersen) [reply components/mpas-ocean/docs/user-guide/index.md:145]: done, thanks.

Comment (mark-petersen) [inline components/mpas-ocean/docs/user-guide/index.md:145]:
```suggestion
- check whether the variable of interest is included in the `streams.ocean` file. Search for the `var name`, or the `var_array` and `var_struct` if applicable. If it is, copy the variable line from other streams into the stream of interest. If it is not included, copy it from the Registry.xml.
```

**Review by mark-petersen — APPROVED** [review]

**Comment by mark-petersen** [comment]
All suggestions by @xylar and @vanroekel have been added. Large pdf file has been removed from history. I squashed the commits to make that easier to remove.

@rljacob and @jonbob this is now ready to merge (before the tutorial!)

**Review by rljacob — APPROVED** [review]

**Comment by jonbob** [comment]
merged to next

**Comment by jonbob** [comment]
merged to master

**Comment by rljacob** [comment]
A few things had to be edited above this for the docs to appear with rest.  I added that in https://github.com/E3SM-Project/E3SM/pull/6374

**Pull request merged**